crates/fuel-core-nats/src/lib.rs

```diff
@@ -138,89 +272,78 @@ pub async fn nats_publisher(
-                } = r
-                {
-                    info!("NATS Publisher: Log Data Length: {}", data.len());
-                    // 0x0000000012345678
-                    let header = vec![0, 0, 0, 0, 18, 52, 86, 120];
-                    if data.starts_with(&header) {
-                        let data = &data[header.len()..];
-                        let mut topics = vec![];
-                        for i in 0..NUM_TOPICS {
-                            let topic_bytes: Vec<u8> = data[32 * i..32 * (i + 1)]
-                                .iter()
-                                .cloned()
-                                .take_while(|x| *x > 0)
-                                .collect();
-                            let topic =
-                                String::from_utf8_lossy(&topic_bytes).into_owned();
-                            if !topic.is_empty() {
-                                topics.push(topic);
-                            }
+            // Publish LogData topics, if any.
+            if let Receipt::LogData {
+                data: Some(data), ..
+            } = r
+            {
+                info!("NATS Publisher: Log Data Length: {}", data.len());
+                // 0x0000000012345678
+                let header = vec![0, 0, 0, 0, 18, 52, 86, 120];
+                if data.starts_with(&header) {
+                    let data = &data[header.len()..];
+                    let mut topics = vec![];
+                    for i in 0..NUM_TOPICS {
+                        let topic_bytes: Vec<u8> = data[32 * i..32 * (i + 1)]
+                            .iter()
+                            .cloned()
+                            .take_while(|x| *x > 0)
+                            .collect();
+                        let topic = String::from_utf8_lossy(&topic_bytes).into_owned();
+                        if !topic.is_empty() {
+                            topics.push(topic);
                         }
-                        let topics = topics.join(".");
-                        // TODO: JSON payload to match other topics? {"data": payload}
-                        let payload = data[NUM_TOPICS * 32..].to_owned();
-
-                        // Publish
-                        info!("NATS Publisher: Receipt#{height}.{contract_id}.{topics}");
-                        jetstream
-                            .publish(
-                                format!("receipts.{height}.{contract_id}.{topics}"),
-                                payload.into(),
-                            )
-                            .await?;
                     }
+                    let topics = topics.join(".");
+                    // TODO: JSON payload to match other topics? {"data": payload}
+                    let payload = data[NUM_TOPICS * 32..].to_owned();
+
+                    // Publish receipt topics
+                    info!("NATS Publisher: Receipt#{height}.{contract_id}.{topics}");
+                    self.publish(
```

Comment:
Same here: Would it be cleaner to extract `publish_receipt` into a separate function?